### PR TITLE
Remove all APIs that use framework fragments

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@samtstern The Android team announced their decision to kill all framework fragments so this PR deprecates all APIs that use those. It includes eventually getting rid of support for normal activities in some cases. Do you think that's okay?

### CODE REVIEW

File: easypermissions/src/main/java/pub/devrel/easypermissions/helper/ActivityPermissionHelper.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 /**
  * Permissions helper for {@link Activity}.
  */
+@Deprecated
 class ActivityPermissionHelper extends BaseFrameworkPermissionsHelper<Activity> {
```
Comment (SUPERCILEX) [inline easypermissions/src/main/java/pub/devrel/easypermissions/helper/ActivityPermissionHelper.java]: This class uses the framework fragments so I deprecated it... not sure if we actually want to do that though.

**Comment by samtstern** [comment]
@SUPERCILEX thanks for bringing this up.  It's an interesting move for sure, maybe we should wait and see what some other prominent Android libraries do to reflect the change?

**Comment by SUPERCILEX** [comment]
@samtstern So I've been poking around [a](https://github.com/Karumi/Dexter) [bunch](https://github.com/ParkSangGwon/TedPermission) [of](https://github.com/tbruyelle/RxPermissions) [libraries](https://github.com/permissions-dispatcher/PermissionsDispatcher), and it seems like we're one of the few who support fragments and the only ones who support framework fragments besides https://github.com/permissions-dispatcher/PermissionsDispatcher. I've created https://github.com/permissions-dispatcher/PermissionsDispatcher/issues/445 to see what they think.

**Comment by samtstern** [comment]
@SUPERCILEX wow thanks for doing the research!

**Comment by SUPERCILEX** [comment]
@samtstern Ok, so Glide and the other permission library have both deprecated framework fragments so I think that's the way to go. 👍

**Comment by samtstern** [comment]
@SUPERCILEX thanks for the research!  I still want to wait until the actual Fragment deprecation is visible in the SDK (it isn't yet, right) to actually release this.

**Comment by SUPERCILEX** [comment]
Alrighty, SGTM

**Comment by SUPERCILEX** [comment]
@samtstern Ok, API 28 is out! 😁 I nuked all the framework APIs so we can jump straight to v2.0, but I'm still on the fence about the `Activity` stuff. Should we support plain activities or require a `FragmentActivity`?

File: easypermissions/src/main/java/pub/devrel/easypermissions/RationaleDialogFragment.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * {@link DialogFragment} to display rationale for permission requests when the request comes from
  * a Fragment or Activity that can host a Fragment.
  */
+@Deprecated
```
Comment (samtstern) [inline easypermissions/src/main/java/pub/devrel/easypermissions/RationaleDialogFragment.java]: What's the reason for marking this internal class `@Deprecated`?  Just for our own knowledge?

**Comment by SUPERCILEX** [comment]
@samtstern SGTM, that's what I was thinking too. As for support lib stability, it's an RC and they said APIs aren't going to change so I don't think we have too much to worry about there. (Though I don't mind waiting a bit longer. 😄)

File: easypermissions/src/main/java/pub/devrel/easypermissions/RationaleDialogFragment.java (hunk 2 above)
Comment (SUPERCILEX) [reply easypermissions/src/main/java/pub/devrel/easypermissions/RationaleDialogFragment.java]: Nope, that was just for me to mark stuff that used the basic `Activity`. Gonna need to refactor a bit.

**Comment by SUPERCILEX** [comment]
Alrighty, this PR is ready to rock whenever you are! 😁

**Comment by samtstern** [comment]
Nice!  I'll give it the respect it deserves on Monday.  Also I am not worried about support lib stability, more worried about forcing our consumers to pull an `rc` dependency into their app.  People tend not to like that.

So I think we might want to wait to release until then.  That said, can still review and merge in the meantime.

**Comment by SUPERCILEX** [comment]
Oh yeah, makes sense. 👍 The support lib team usually ships a final version pretty fast after an RC, so I doubt we'll be waiting too long. 😄

**Comment by SUPERCILEX** [comment]
Yup. We can also get rid of the few remaining deprecated methods.

**Pull request merged**